ambry-router/src/main/java/com/github/ambry/router/DecryptJob.java

```diff
@@ -110,7 +115,12 @@ public void run() {
    */
   @Override
   public void closeJob(GeneralSecurityException gse) {
-    callback.onCompletion(null, gse);
+    if (closed.compareAndSet(false, true)) {
+      if (encryptedBlobContent != null) {
+        encryptedBlobContent.release();
+      }
```

Comment:
Let's do ReferenceCountUtil.safeRelease here since to avoid the case the encryptedblobcontent is already released in different place.